clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -129,16 +129,12 @@ public boolean isComplete() {
 
     @Override
     public byte[] unwrap(byte[] incoming, int offset, int len) {
-        if (!isComplete())
-            throw new IllegalStateException("Authentication exchange has not completed");
-        return Arrays.copyOfRange(incoming, offset, offset + len);
+        throw new IllegalStateException("Mechanism does not support integrity nor confidentality");
```

Comment:
Should we perhaps alter the message within `IllegalStateException` depending on whether the authentication is completed or not and add the name of the mechanism as well? Similar to how the JDK implementation does it for [PlainClient](https://github.com/openjdk/jdk21/blob/890adb6410dab4606a4f26a942aed02fb2f55387/src/java.security.sasl/share/classes/com/sun/security/sasl/PlainClient.java#L144)

Reply:
Thanks, re-written to match the JDK behaviour and be more consistent with the previous code.